test/kitchen/docs/README.md

```diff
@@ -1,42 +1,128 @@
+
+These files are `.erb` template files, the rendered contents (eg. the list of platforms) depend on what variables are passed to the `run-test-kitchen.sh` script.
+
+#### Test execution (tasks/run-test-kitchen.sh)
+
+Each individual test environment is controlled from the `.gitlab-ci.yml` jobs. The tests are set up using environment variables:
+
+`TEST_PLATFORMS`: a list of test platforms which are turned into test kitchen platforms. In `.gitlab-ci.yml`, the platform list has the following syntax:
+        `short_name1,type1,azure_fully_qualified_name1|short_name2,type2,azure_fully_qualified_name2`
+
+There are two types of platforms:
+- ones which use a public image (type `urn`): the azure qualified name for such an image can be found by using `az vm image list` to search for the image (using the publisher name, the offer name and/or the sku). The `urn` field contains the azure fully qualified name.
+- ones which use a private image (type `id`): the image is in an image gallery in the `kitchen-test-images` resource group of the Azure account. The fully qualified name is the full path to the image in the gallery.
+
+We mainly use private images for platforms we still want to test, but have been removed from the public Azure images list (eg. because they're EOL).
+
+**Note:** this variable is set by the `kitchen_os_X` jobs.
+
+`AZURE_LOCATION`: a string that indicates on which Azure region the test VMs are spun up.
+
+**Note:** this variable is set by the `kitchen_azure_location_X` jobs.
+
+`AGENT_FLAVOR`: a string that indicates which flavor the Agent is installed (eg. datadog-agent, datadog-iot-agent). In particular, it helps knowing which package has to be installed, and which tests should be applied (eg. doing Python tests on the IoT Agent doesn't make sense).
+
+**Note:** this variable is set by the `kitchen_datadog_X_flavor` jobs.
+
+This configuration is fed to the `tasks/run-test-kitchen.sh`. That script also takes two mandatory parameters:
+
+- an identifier for the test suite that is run (eg. `install-script`, `upgrade5`),
+- the major Agent version that is tested (eg. `6` or `7`).
+
+In turn, it sets up additional variables for Azure (authentication/environment variables), as well as some variables used during testing (such as the exact expected Agent version, gathered from the `inv agent.version` command).
+
+It concatenates the `kitchen-azure-common.yml` file with the file specific to the provided test suite into a `kitchen.yml` file, and then executes the actual kitchen command.
 
 ### Test Implementation
 
 #### Directory structure
 
-Test kitchen imposes a very specific directory structure.  For more information, see the test kitchen documentation.  However, a brief overview:
+Test kitchen imposes a very specific directory structure. For more information, see the test kitchen documentation. However, a brief overview:
```

Comment:
nit:
```suggestion
Test kitchen imposes a very specific directory structure. For more information, see the [test kitchen documentation](https://kitchen.ci/docs). However, a brief overview:
```
I think that is the right link but check it just in case.